libevmasm/RuleList.h

```diff
@@ -681,6 +681,22 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart8(
 			// X - (A + Y) -> (X - Y) + (-A)
 			Builtins::SUB(X, Builtins::ADD(A, Y)),
 			[=]() -> Pattern { return Builtins::ADD(Builtins::SUB(X, Y), 0 - A.d()); }
+		}, {
+			// (X - A) - Y -> (X - Y) - A
+			Builtins::SUB(Builtins::SUB(X, A), Y),
+			[=]() -> Pattern { return Builtins::SUB(Builtins::SUB(X, Y), A); }
```

Comment:
Will this rule ever be matched rather than first hitting ``X - A -> X + (-A)``? I'd need to look through the matching mechanism to be able to tell for sure, but maybe a few test cases for these would be nice in general.
We should probably in general have `isoltest`-tests for single and multiple applications of these rules, but may be overkill for this...

Reply:
Without dedicated rule list tests, probably hard to test, since
``X - A`` -> ``X + (-A)``
followed by
``(X + (-A)) - Y -> (X - Y) + (-A)``
will be indistinguishable from

``(X - A) - Y) -> (X - Y) - A``
followed by
``(X - Y) - A -> (X - Y) + (-A)``

But in that sense this rule is probably redundant, isn't it?

Reply:
Yes, some rules are redundant, but I think I need them anyway because of the way the KnowledgeBase works.

Reply:
Actually no, they are really redundant.